lib/validation.js

```diff
@@ -109,6 +109,39 @@ module.exports = function (yargs, usage) {
     }
   }
 
+  // validate arguments limited to enumerated choices
+  self.limitedChoices = function (argv) {
+    var options = yargs.getOptions()
+    var invalid = {}
+
+    if (!Object.keys(options.choices).length) return
+
+    Object.keys(argv).forEach(function (key) {
+      if (key !== '$0' && key !== '_' &&
+        options.choices.hasOwnProperty(key)) {
+        [].concat(argv[key]).forEach(function (value) {
+          // TODO case-insensitive configurability
```

Comment:
I could go either way; I think in the most common case people will tend to pass in all lower case options -- so it might be overkill to make this configurable out of the gate.

I'm fine with leaving the TODO, let's see if anyone requests this feature and makes a good case for it?